flax/linen/module.py

```diff
@@ -1104,6 +1104,17 @@ def other_fn(instance, ...):
     if method is None:
       method = self.__call__
     method = _get_unbound_fn(method)
+    # Try to detect if user accidentally passed {'params': {'params': ...} or
+    # {'kernel': ...} as `variables`.
+    if 'params' in variables and isinstance(
```

Comment:
I think you can remove the check for `FrozenDict`. `variables` is not required to be a FrozenDict either, so I think this error should also be raised if it is a regular dict.

Reply:
if i check only `isinstance(variables['params'], dict), it won't catch flax.core.frozen_dict.FrozenDict since FrozenDict is not a dict, so the test would fail. If a real user would ever pass a FrozenDict, how about `isinstance(variables['params'], (dict, FrozenDict))`?